qiskit/transpiler/__init__.py

```diff
@@ -19,8 +19,403 @@
 
 .. currentmodule:: qiskit.transpiler
 
+Overview
+========
+Transpilation is the process of rewriting a given input circuit to match
+the topoplogy of a specific quantum device, and/or to optimize the circuit
+for execution on present day noisy quantum systems.
+
+Most circuits must undergo a series of transformations that make them compatible with
+a given target device, and optimize them to reduce the effects of noise on the
+resulting outcomes.  Rewriting quantum circuits to match hardware constraints and
+optimizing for performance can be far from trivial.  The flow of logic in the rewriting
+tool chain need not be linear, and can often have iterative sub-loops, conditional
+branches, and other complex behaviors.  That being said, the basic building blocks
+follow the structure given below:
+
+.. image:: /source_images/transpiling_core_steps.png
+
+.. raw:: html
+
+   <br><br>
+
+
+Supplementary Information
+=========================
+
+.. container:: toggle
+
+   .. container:: header
+
+      **Basis Gates**
+
+   When writing a quantum circuit you are free to use any quantum gate (unitary operator) that
+   you like, along with a collection of non-gate operations such as qubit measurements and
+   reset operations.  However, when running a circuit on a real quantum device one no longer
+   has this flexibility.  Due to limitations in, for example, the physical interactions
+   between qubits, difficulty in implementing multi-qubit gates, control electronics etc,
+   a quantum computing device can only natively support a handful of quantum gates and non-gate
+   operations. In the present case of IBM Q devices, the native gate set can be found by querying
+   the devices themselves, and looking for the corresponding attribute in their configuration:
+
+   .. jupyter-execute::
+      :hide-code:
+      :hide-output:
+
+      from qiskit.test.mock import FakeVigo
+      backend = FakeVigo()
+
+   .. jupyter-execute::
+
+      backend.configuration().basis_gates
+
+   We see that IBM Quantum devices support five native gates: four single-qubit gates
+   [`u1`, `u2`, `u3`, and `id`] and one two-qubit entangling gate `cx`.  In addition, the
+   devices support qubit measurements (otherwise we can not read out an answer!).
```

Comment:
This whole section that goes into detail about u1,u2,u3,cx is a bit orthogonal to what the transpiler is. Those gates are already documented, here I think we should go into detail about different transpiler passes. Besides, these u gates are going to go obsolete soon: https://github.com/Qiskit/qiskit-terra/issues/4106

Reply:
Ok will cut out. Not sure how to reconcile the desire to talk about transpiler passes here with exposing them into the docs.  Surely the passes module should discuss this.  I can however link to the docs there and then the users would know exactly what is going on when someone gets around to making those docs.

Reply:
Passes are also already at top-level in the docs.